admin-dev/themes/new-theme/js/pages/carrier/form/carrier-form-manager.ts

```diff
@@ -82,6 +86,35 @@ export default class CarrierFormManager {
     this.$shippingMethodInput.on('change', () => this.refreshCurrentShippingSymbol());
     $(CarrierFormMap.zonesContainer).on('click', CarrierFormMap.deleteZoneButton, (e:Event) => this.onDeleteZone(e));
     this.eventEmitter.on(CarrierFormEventMap.rangesUpdated, (ranges: Range[]) => this.onChangeRanges(ranges));
+
+    this.initErrorListener();
+  }
+
+  private initErrorListener(): void {
+    const carrierForm = document.querySelector(CarrierFormMap.form);
+    const requiredFields = carrierForm?.querySelectorAll('[required]');
```

Comment:
Do we need do filter only inputs that are required? There may be other inputs with min/max value or other stuff that are enforced natively in HTML. So we could check every inputs and listen to their invalid event?